app/components/progress-bar.js

```diff
@@ -0,0 +1,47 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { debounce } from '@ember/runloop';
+
+export default class ProgressBarComponent extends Component {
+  @tracked isEditable = false;
+  @tracked value = this.args.value;
+  lastEditTime = null;
+
+  @action turnEditModeOn() {
+    this.isEditable = true;
+    this.lastEditTime = Date.now();
+    this.setEditableToFalse();
+  }
+
+  setEditableToFalse() {
+    setTimeout(() => {
+      const timeDelta = Date.now() - this.lastEditTime;
+      if (this.isEditable && timeDelta >= 5000) {
+        this.isEditable = false;
+      } else if (this.isEditable) {
+        this.setEditableToFalse();
+      }
+    }, 5000);
+  }
```

Comment:
- why are we adding this delay ?

Reply:
When we click the edit button the progress bar becomes editable and edit button shows progress percent. And if user doesn't perform any actions on the progress bar then this timer reverts everything to original state.